bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -675,19 +675,42 @@ type DiskSetup struct {
 	Filesystems []Filesystem `json:"filesystems,omitempty"`
 }
 
+// DiskLayout represents an array of partition specifications
+type DiskLayout []PartitionSpec
+
+// PartitionSpec defines the size and optional type for a partition
+type PartitionSpec struct {
+	// Percentage of disk that partition will take (1-100)
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=100
+	Percentage int32 `json:"percentage"`
+
+	// PartitionType is the numerical value of the partition type (optional)
+	// +optional
+	PartitionType *int32 `json:"partitionType,omitempty"`
```

Comment:
What are the valid values for this value? And how would I, as an end user, understand those? 

Is there a limit on this? Is `0` a valid value? Are negative values valid?

Reply:
Actually I can use a string here to reflect the correct  linux [partition types](https://www.system-rescue.org/disk-partitioning/Partitions-attributes/). It was an attempt to purely map the disk layout as described in the cloud init docs.

Reply:
There's a choice to make here. Either you can allow all of the valid values there, in which case this should be a 2 character lowercase hex value. Or, you can allow specific, common types, and do that by creating an Enum here and creating PascalCase aliases to the formats we actually want to support.

IMO, the latter provides a better UX